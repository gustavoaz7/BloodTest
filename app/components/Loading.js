import React from 'react';
import { View, ActivityIndicator } from 'react-native';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import { theme } from '../theme';

export default function Loading(props) {
  const { size, color } = props;
  return (
    <Container>
      <ActivityIndicator size={size} color={color} />
    </Container>
  );
}

const Container = styled(View)`
  flex: 1;
  align-items: center;
  justify-content: center;
`;

Loading.propTypes = {
  size: PropTypes.oneOf(['small', 'large']),
  color: PropTypes.string,
};

Loading.defaultProps = {
  size: 'large',
  color: theme.primary,
};
