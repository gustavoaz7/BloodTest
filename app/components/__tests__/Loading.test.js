import React from 'react';
import { shallow } from 'enzyme';
import Loading from '../Loading';

describe('Loading', () => {
  it('renders without crashing', () => {
    shallow(<Loading />);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Loading />);
    expect(wrapper).toMatchSnapshot();
  });
});
