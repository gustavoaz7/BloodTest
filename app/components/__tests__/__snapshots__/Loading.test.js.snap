// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading matches snapshot 1`] = `
<Styled(View)>
  <ActivityIndicator
    animating={true}
    color="#f5785a"
    hidesWhenStopped={true}
    size="large"
  />
</Styled(View)>
`;
